lib/_http_client.js

```diff
@@ -230,9 +230,7 @@ function ClientRequest(input, options, cb) {
   const headersArray = ArrayIsArray(options.headers);
   if (!headersArray) {
     if (options.headers) {
-      const keys = ObjectKeys(options.headers);
-      for (let i = 0; i < keys.length; i++) {
-        const key = keys[i];
+      for (const key of ObjectKeys(options.headers)) {
```

Comment:
I do not think we should make this change for performance reasons.

Reply:
I'll revert these changes and add comment about the performance - so that future devs know about it.

Is there a benchmarks link which I can share detailing performance concerns of for...of loop?

Reply:
There isn't, this comes from experience.

Reply:
Resolved in https://github.com/nodejs/node/pull/30958/commits/8d435801647cc57a6b14be4b484c1bb3de781413